test/config/test/projects.test.ts

```diff
@@ -138,3 +138,60 @@ it('fails if workspace is filtered by the project', async () => {
     "./vitest.config.js"
 ].`)
 })
+
+describe('the config file names', () => {
+  it('[glob] the name has "unit" between "vitest" and "config" and works', async () => {
+    const { exitCode } = await runInlineTests({
+      'vitest.unit.config.js': {},
+      'vitest.config.js': {
+        test: {
+          passWithNoTests: true,
+          projects: ['./vitest.*.config.js'],
+        },
+      },
+    })
```

Comment:
I would rather refactor these test setups to not load same configuration twice. So root configuration that has `projects` would not be considered a project.

```suggestion
    const { exitCode } = await runInlineTests({
      // Project configs that are being tested
      'vitest.unit.config.js': {},
      'vitest.config.js': {
        test: { passWithNoTests: true },
      },
    },
    // Root config that sets the pattern
    { projects: ['./vitest.*.config.js'] })
```

Reply:
I am confused what do you mean, `vitest.*.config.js` does not match `vitest.config.js`

Reply:
This doesn't consider the root config as project too?

https://github.com/sheremet-va/vitest/blob/38c25d93edfa87c0d15769b294932116c310f97e/test/config/test/projects.test.ts#L158-L164

```ts
      'unit.config.js': {},
      'vitest.config.js': {
        test: {
          projects: ['./*.config.js'],
        },
      },
```

Reply:
It would match it, but it doesn't matter because it also catches `unit.config.js` first and fails without loading any configs

Reply:
Sure. Overall the comment is about using inline options for `projects` pattern, and configuration files for files that are being tested. Mixing both seemed confusing at first.